ast/ast.go

```diff
@@ -130,6 +130,20 @@ func (ms *MovementStatement) statementNode() {}
 // TokenLiteral returns a string representation of the movement statement.
 func (ms *MovementStatement) TokenLiteral() string { return ms.Token.Literal }
 
+// MartStatement is a Poryscript mart statement.
```

Comment:
I noticed the parser code for `mart` and `movement` are essentially identical, minus the fact that `movement` supports using `*` to repeat movement commands.  Rather than duplicating all of this, including poryswitch parsing, I think it would be better to create a new higher-order concept of a "Data List Statement", which would handle parsing both `MartStatement` and `MovementStatement`.  Maybe this would only exist as a set of parsing functions, and not as a literal struct in `ast.go`.  Does that make sense?

Reply:
That makes complete sense. I would probably do something along the lines of a common function that returns a string array, with the `*` behavior be optional via a boolean argument (as the syntax could be useful in the future; although unlikely, any other data list in the future could easily hook into the "duplicate this list element" behavior in this way). There would still be a mart-specific parser function that initializes the struct data, but it only goes one layer deep, because the `MartItems` assignment would simply use the new common function. Same goes for MovementData.

However, if you see a more elegant solution, I'd be happy to do that instead.

Reply:
That sounds reasonable, I think.  I'll probably have some more opinions after it's been implemented.